### PR TITLE
# noqa removal in code base

### PR DESCRIPTION
**Pull request by kaathi524**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by kaathi524** [comment]
As I removed all the `# noqa: F401`, Build failed "error F401: '*' imported but unused"

**Comment by jbrockmendel** [comment]
Some of these are needed to pass our linting.  See the output in the "Checks" CI build.

That said, there are some of these (particularly non-specific "noqa"s) that are leftover from olden days and cleaning them up is definitely helpful.

**Comment by datapythonista** [comment]
Where is this coming from? Do you have an issue number, or a discussion in some other PR or issue that you want to reference in the description, so we know the context?

**Comment by WillAyd** [comment]
I don't think we can remove all of these - looks to be some failing CI you can keep around. I'm surprised to only see 10 in CI but maybe that's a good thing...

**Comment by WillAyd** [comment]
@datapythonista this is a follow up to #29209 and particularly my comment https://github.com/pandas-dev/pandas/pull/29209#issuecomment-546045353

File: asv_bench/benchmarks/algorithms.py [hunk 1]
```diff
@@ -183,4 +183,4 @@ def time_argsort(self, N):
         self.array.argsort()
 
 
-from .pandas_vb_common import setup  # noqa: F401 isort:skip
+from .pandas_vb_common import setup
```
Comment (WillAyd) [inline asv_bench/benchmarks/algorithms.py]: These can't be deleted and will still cause errors

**Review by WillAyd — CHANGES_REQUESTED** [review]
@kaathi524 to be clear you can't just blanket remove all of these. If you run flake8 locally you will see a lot of failures doing this.

Instead of doing this all at once you can also split into a few PRs to make things easier

File: asv_bench/benchmarks/algorithms.py (hunk 1 above)
Comment (kaathi524) [reply asv_bench/benchmarks/algorithms.py]: Okay. Will split and fix. @WillAyd

**Comment by jbrockmendel** [comment]
@kaathi524 can this be closed in favor of several smaller PRs?

**Comment by kaathi524** [comment]
> @kaathi524 can this be closed in favor of several smaller PRs?

Yeah sure. I tried running flake8 in my local setup without removing any 'noqa' but it throwing error.

**Comment by datapythonista** [comment]
Thanks @kaathi524, looking forward to the fixes in the smaller PRs. Please reference this PR when you open them, so we can track the discussion.

**Pull request closed without merging**